hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractMultiValueGetter.java

```diff
@@ -156,15 +156,23 @@ private Object getParentObject(Object obj) throws Exception {
 
     private void reduceArrayInto(MultiResult collector, Object[] currentObject) {
         Object[] array = currentObject;
-        for (int i = 0; i < array.length; i++) {
-            collector.add(array[i]);
+        if (array.length == 0) {
```

Comment:
it's for this test case
```
    @Test
    public void comparable_primitive_reduced_atLeaf_comparedToNull_matching() {
        execute(Input.of(BOND, KRUEGER),
                Query.of(equal("limbs_[any].tattoos_[any]", null), mv),
                Expected.of(BOND, KRUEGER));
    }
```
Both input object have empty tattoos array/list.
they query would work fine if you changed `any` to a concrete `number`. With any it didn't return a match with null and it should